subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -149,7 +149,7 @@ int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
 	/* Check if the status field(1st field) of the object has been parsed */
 	if (ret & 0x01) {
 		memcpy(status, rsp.status,
-			MIN(strlen(rsp.status), STATUS_MAX_LEN));
+			MIN(strlen(rsp.status) + 1, STATUS_MAX_LEN));
```

Comment:
In case `strlen(rsp.status) + 1` > `STATUS_MAX_LEN`, `status` will also be non NULL-terminated. Perhaps we should add `status[STATUS_MAX_LEN - 1] = '\0'` to be on the safe side?

Reply:
Good point, this applies to all of the strings being copied right?

I will add a test case which verifies this as well.

Reply:
Yes, I should've mentioned that this should be applied all below.

Reply:
Should be fixed now, would be good if you checked how its done.